File: 1493-longest-subarray-of-1s-after-deleting-one-element/1493-longest-subarray-of-1s-after-deleting-one-element.py
class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        k, i, j = 1, 0, 0
        for j in range (len(nums)):
            if nums[j] == 0: 
                k -= 1
            if k < 0 and nums[i] == 1:
                i += 1
            elif k < 0 and nums[i] == 0:
                i += 1
                k += 1
        return j - i

File: 1493-longest-subarray-of-1s-after-deleting-one-element/1493-longest-subarray-of-1s-after-deleting-one-element.java
class Solution {
    public int longestSubarray(int[] nums) {
        int k = 1, i = 0, j;
        for(j = 0; j < nums.length; ++j) {
            if(nums[j] == 0) --k;
            if(k < 0 && nums[i++] == 0) ++k;
        }
        return j - i - 1;
    }
}